File: src/Entity/Reservation.php
<?php

namespace App\Entity;

use App\Repository\ReservationRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ReservationRepository::class)]
class Reservation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    
    
    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Le titre ne peut pas être vide')]
    #[Assert\Length(max: 255, maxMessage: 'Le titre ne peut pas dépasser {{ limit }} caractères')]
    #[Assert\Regex(
        pattern: '/^[a-zA-Z\- ]+$/',
        message: 'Le nom ne peut contenir que des lettres, des tirets et des espaces'
        )]
    private ?string $nom = null;

    /**
 * @Assert\NotBlank(message="L'email est obligatoire")
 * @Assert\Email(
 *     message = "L'email '{{ value }}' n'est pas valide.",
 *     mode = "html5"
 * )
 * @ORM\Column(type="string", length=255)
 */

    #[ORM\Column(length: 255)]
    private ?string $email = null;
    /**
     * @Assert\NotBlank(message="La date de réservation est obligatoire.")
     * @ORM\Column(type=Types::DATE_MUTABLE)
     */

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    private ?\DateTimeInterface $date = null;

    /**
     * @Assert\NotBlank(message="Le nombre de personnes est obligatoire.")
     * @Assert\Positive(message="Le nombre de personnes doit être positif.")
     * @ORM\Column
     */

    #[ORM\Column]
    public ?int $nbr_personne = null;

    #[ORM\ManyToOne(inversedBy: 'reservations')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Restaurant $restaurant = null;

    #[ORM\ManyToOne(inversedBy: 'reservations')]
    #[ORM\JoinColumn(nullable: false)]
    private ?User $user_id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): static
    {
        $this->email = $email;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getNbrPersonne(): ?int
    {
        return $this->nbr_personne;
    }

    public function setNbrPersonne(int $nbr_personne): static
    {
        $this->nbr_personne = $nbr_personne;

        return $this;
    }

    public function getRestaurant(): ?Restaurant
    {
        return $this->restaurant;
    }

    public function setRestaurant(?Restaurant $restaurant): static
    {
        $this->restaurant = $restaurant;

        return $this;
    }

    public function getUserId(): ?User
    {
        return $this->user_id;
    }

    public function setUserId(?User $user_id): static
    {
        $this->user_id = $user_id;

        return $this;
    }
}

File: src/main/java/Entity/Reservation.java
package Entity;

import javafx.beans.property.IntegerProperty;
import javafx.beans.property.SimpleIntegerProperty;
import javafx.beans.property.StringProperty;
import javafx.beans.property.SimpleStringProperty;

public class Reservation {
    private int id;
    private IntegerProperty restaurantId;
    private StringProperty nom;
    private StringProperty email;
    private StringProperty date;
    private int nbrPersonne;
    private final IntegerProperty reservationId;
    private Restaurant selectedRestaurant;

    public Restaurant getSelectedRestaurant() {
        return selectedRestaurant;
    }

    public void setSelectedRestaurant(Restaurant selectedRestaurant) {
        this.selectedRestaurant = selectedRestaurant;
    }
    private Restaurant restaurant;

    public Restaurant getRestaurant() {
        return restaurant;
    }

    public void setRestaurant(Restaurant restaurant) {
        this.restaurant = restaurant;
    }

    // Constructors
    public Reservation(int reservationId, int restaurantId) {
        this.reservationId = new SimpleIntegerProperty(reservationId);
        this.restaurantId = new SimpleIntegerProperty(restaurantId);
        this.nom = new SimpleStringProperty("");
        this.email = new SimpleStringProperty("");
        this.date = new SimpleStringProperty(""); // Initialize date property
    }


    public Reservation(int reservationId, int restaurantId, String nom, String email, String date, int nbrPersonne) {
        this.reservationId = new SimpleIntegerProperty(reservationId);
        this.restaurantId = new SimpleIntegerProperty(restaurantId);


        this.nom = new SimpleStringProperty(nom);
        this.email = new SimpleStringProperty(email);
        this.date = new SimpleStringProperty(date);
        this.nbrPersonne = nbrPersonne;
    }

    public Reservation(int selectedRestaurantId, String nom, String email, String date, int nbrPersonne, IntegerProperty reservationId) {
        this.reservationId = reservationId;
        this.date = new SimpleStringProperty(date); // Initialize date property
    }

    // Getters and setters for properties
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getRestaurantId() {
        return restaurantId.get();
    }

    public void setRestaurantId(int restaurantId) {
        this.restaurantId.set(restaurantId);
    }

    public IntegerProperty restaurantIdProperty() {
        return restaurantId;
    }

    public String getNom() {
        return nom.get();
    }

    public void setNom(String nom) {
        this.nom.set(nom);
    }

    public StringProperty nomProperty() {
        return nom;
    }

    public String getEmail() {
        return email.get();
    }

    public void setEmail(String email) {
        this.email.set(email);
    }

    public StringProperty emailProperty() {
        return email;
    }

    public String getDate() {
        return date.get();
    }

    public void setDate(String date) {
        this.date.set(date);
    }

    public StringProperty dateProperty() {
        return date;
    }

    public int getNbrPersonne() {
        return nbrPersonne;
    }

    public void setNbrPersonne(int nbrPersonne) {
        this.nbrPersonne = nbrPersonne;
    }

    public IntegerProperty NbrPersonneProperty() {
        return new SimpleIntegerProperty(nbrPersonne);
    }

    public int getReservationId() {
        return reservationId.get();
    }

    public void setReservationId(int reservationId) {
        this.reservationId.set(reservationId);
    }

    public IntegerProperty reservationIdProperty() {
        return reservationId;
    }
}
